state/status.go

```diff
@@ -199,49 +199,63 @@ type statusHistoryArgs struct {
 	filter    status.StatusHistoryFilter
 }
 
-func statusHistory(args *statusHistoryArgs) ([]status.StatusInfo, error) {
-	filter := args.filter
-	if err := args.filter.Validate(); err != nil {
-		return nil, errors.Annotate(err, "validating arguments")
-	}
-	statusHistory, closer := args.st.getCollection(statusesHistoryC)
-	defer closer()
-
+// fetchNStatusResults will return status for the given key filtered with the
+// given filter or error.
+func fetchNStatusResults(col mongo.Collection, key string,
+	filter status.StatusHistoryFilter) ([]historicalStatusDoc, error) {
 	var (
 		docs  []historicalStatusDoc
 		query mongo.Query
 	)
-	baseQuery := bson.M{"globalkey": args.globalKey}
+	baseQuery := bson.M{"globalkey": key, "statusinfo": bson.M{"$ne": "running update-status hook"}}
```

Comment:
This code still doesn't use the filter.Exclude parameter.
It's also a bit messy with the cut and paste repetition.